chainer/functions/math/minmax.py

```diff
@@ -75,6 +77,56 @@ def _fwd(self, x, xp):
         return xp.amin(x, axis=self.axis, keepdims=self.keepdims)
 
 
+class IndexSelectorBase(function.Function):
+    """Select index of an array element from a given axis."""
+
+    def __init__(self, axis=None, keepdims=False):
+        if axis is None:
+            self.axis = None
+        elif isinstance(axis, int):
+            self.axis = axis
+        else:
+            raise TypeError('None or int are required')
+
+    def _fwd(self, x, xp):
+        raise NotImplementedError('_fwd should be implemented in sub-class.')
+
+    def check_type_forward(self, in_types):
+        type_check.expect(
+            in_types.size() == 1,
+            in_types[0].dtype.kind == 'f'
+        )
+
+        if self.axis is not None:
+            if self.axis >= 0:
+                type_check.expect(
+                    self.axis < in_types[0].ndim,
+                )
+            else:
+                type_check.expect(
+                    -self.axis - 1 < in_types[0].ndim,
+                )
+
+    def forward(self, x):
+        xp = cuda.get_array_module(*x)
+        return xp.asarray(self._fwd(x[0], xp)),
+
+    def backward(self, x, gy):
```

Comment:
Do we need backward? I think this function is not differentiable, thus it should return None in backward, which is the default behavior of the base class `Function`.